nixos/modules/services/system/nix-daemon.nix

```
@@ -233,6 +242,9 @@ in
         LimitNOFILE = 1048576;
         Delegate = "yes";
         DelegateSubgroup = "supervisor";
+      }
+      // lib.optionalAttrs cfg.vacuumDB {
+        ExecStartPre = "-${lib.getBin pkgs.sqlite}/bin/sqlite3 /nix/var/nix/db/db.sqlite VACUUM";
```

```suggestion
        ExecStartPre = "-sqlite3 /nix/var/nix/db/db.sqlite VACUUM";
```

`sqlite` is already in the services PATH, so we don't need to refer to its exact path

---

I'm not sure that's the case, I get ```nix-daemon.service: Unable to locate `sqlite3`: No such file or directory``` on the nix-daemon service logs with this change.

---

```suggestion
        ExecStartPre = "-${lib.getExe pkgs.sqlite} /nix/var/nix/db/db.sqlite VACUUM";
```